geo-types/src/private_utils.rs

```diff
@@ -72,14 +72,15 @@ where
     }
     let dx = end.x - start.x;
     let dy = end.y - start.y;
-    let r = ((point.x - start.x) * dx + (point.y - start.y) * dy) / (dx.powi(2) + dy.powi(2));
+    let d = dx * dx + dy * dy;
```

Comment:
I guess this is `d_squared`?

Reply:
I _think_ explicit var * var is faster (or thought to be faster, in a folkloric way) than var.powf(T::one() + T::one()), or certainly was for a long time, but who really knows the compiler's inner thoughts (godbolt does)

Reply:
To be clear, I'm just picking nits about the variable naming.

```
let d_squared = dx * dx + dy * dy;
```

I think d implies:
```
let d = (dx * dx + dy + dy).sqrt()
```

But maybe I'm reading in too deep.

Reply:
Ohhhh sorry

Reply:
Done. Regarding performance, at least in isolation, `x * x` and `x.powi(2)` produce the same code, this is for consistency reasons, since it can make people think it's intentional. I actually tracked the code down to https://github.com/georust/geo/pull/139 to see if there was any discussion of it.